sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuDeviceManager.scala

```diff
@@ -63,8 +64,30 @@ object GpuDeviceManager extends Logging {
 
   @volatile private var poolSizeLimit = 0L
 
+  private class DefaultOverrideProvider {
```

Comment:
I don't see a lot of value in having a class for this.

```
def getSplitUntilSize: Long = {
  val conf = new RapidsConf(SQLConf.get)
  val override = if (conf.isTestEnabled) {
    conf.splitUntilSizeOverride
  } else {
    None
  }
  override.getOrElse(Math.max(poolSizeLimit / 8, 100 * 1024 * 1024))
}
```

Reply:
I'll use your approach for this PR. I think we can the idea in the back of our heads if we find ourselves adding overrides all over the place so we can trigger special cased code, so we have a class that groups them (perhaps near RapidsConf).